src/lib/reasoners/sat_solver_sig.ml

```diff
@@ -72,58 +72,78 @@ module type S = sig
   exception Unsat of Explanation.t
   exception I_dont_know
 
-  (* the empty sat-solver context *)
-  val empty : unit -> t
-  val empty_with_inst : (Expr.t -> bool) -> t
-
-  (** [push env n] add n new assertion levels.
-      A guard g is added for every expression e assumed at the current
-      assertion level.
-      Ie. assuming e after the push will become g -> e,
-      a g will be forced to be true (but not propagated at level 0) *)
+  val empty : ?selector:(Expr.t -> bool) -> unit -> t
+  (** [empty ~selector ()] creates an empty environment.
+
+      The optional argument [selector] is used to filter ground facts
+      discovered by the instantiation engine. *)
+
   val push : t -> int -> unit
+  (** [push env n] adds [n] new assertion levels in [env].
+
+      Internally, for each new assertion level, a fresh guard [g] is created
+      and all formulas [f] assumed at this assertion level is replaced by
+      [g ==> f].
+
+      The guard [g] is forced to be [true] but not propagated at level [0]. *)
 
-  (** [pop env n] remove an assertion level.
-      Internally, the guard g introduced in the push correponsding to this pop
-      will be propagated to false (at level 0) *)
   val pop : t -> int -> unit
+  (** [pop env n] removes [n] assertion levels in [env].
+
+      Internally, the guard [g] introduced in [push] corresponding to this pop
+      is propagated to [false] at level [0]. *)
 
-  (* [assume env f] assume a new formula [f] in [env]. Raises Unsat if
-     [f] is unsatisfiable in [env] *)
   val assume : t -> Expr.gformula -> Explanation.t -> unit
+  (** [assume env f dep] assumes the ground formula [f] in [env].
+      The [dep] argument can be used to generate an unsat core.
+
+      @raise Unsat if [f] is unsatisfiable in [env]. *)
 
   val assume_th_elt : t -> Expr.th_elt -> Explanation.t -> unit
+  (** [assume env f exp] assumes a new formula [f] with the explanation [exp]
+      in the theory environment of [env]. *)
 
-  (* [pred_def env f] assume a new predicate definition [f] in [env]. *)
   val pred_def : t -> Expr.t -> string -> Explanation.t -> Loc.t -> unit
+  (** [pred_def env f] assumes a new predicate definition [f] in [env]. *)
 
-  (** [optimize env ~is_max o] registers the expression [o]
-      as an objective function. *)
   val optimize : t -> is_max:bool -> Expr.t -> unit
+  (** [optimize env ~is_max o] registers the expression [o] as an objective
+      function.
+
+      After optimization, the value of this objective is returned by
+      [get_objectives]. *)
 
-  (* [unsat env f size] checks the unsatisfiability of [f] in
-     [env]. Raises I_dont_know when the proof tree's height reaches
-     [size]. Raises Sat if [f] is satisfiable in [env] *)
   val unsat : t -> Expr.gformula -> Explanation.t
+  (** [unsat env f size] checks the unsatisfiability of [f] in [env].
```

Comment:
There is no `size` argument? `unsat` has only two arguments; this seems outdated.